#157244155 Integrate paystack payment

Felglitzzz opened the pull request:
#### What does this PR do?
Integrates paystack payment as one of the payment options for users

#### Description of Task to be completed?

- create a custom plugin for paystack payment option
- Tweak the checkout files to reflect needful fields
- Integrate the paystack inline API
- Integrate server methods to allow for payments via paystack

#### How should this be manually tested?

- npm run git clone `https://github.com/andela/nazgul-rc.git` to clone the repository
- navigate to the nazgul-rc folder
- run `curl https://install.meteor.com/ | sh`
- run `npm install -g reaction-cli`
- install dependencies with`meteor npm install`
- run `reaction` to start up server
- once app is running, navigate to the app using localhost:3000
- navigate to shop page
- add items to checkout by clicking the cart icon
- enter the needful details, select a shipping rate and fill your details in the form
- click on "Complete your order" to make payment
- follow the payment instructions

#### What are the relevant pivotal tracker stories?
#157245700
#### Screenshots (if appropriate)
<img width="1440" alt="paystack" src="https://user-images.githubusercontent.com/29508652/39712129-d65a84d6-5219-11e8-861f-37099da9edc8.png">
<img width="1440" alt="paystack2" src="https://user-images.githubusercontent.com/29508652/39712131-d68204ca-5219-11e8-9fe9-92e71c769356.png">
<img width="1440" alt="paystack3" src="https://user-images.githubusercontent.com/29508652/39712132-d6a41d4e-5219-11e8-8a1b-e5f0ef0cec99.png">
<img width="1440" alt="paystack4" src="https://user-images.githubusercontent.com/29508652/39712133-d6c8303a-5219-11e8-9b60-ef3118519403.png">

Commit:
feature(payment): add paystack provider in payment provider section
add payment-paystack package as a plugin in imports folder
modify the package to reflect paystack payment provider
modify the schema to reflect the data to be submitted for payment

Commit:
feature(payment): Integrate paystack payment
create payments-paystack custom plugin

Commit:
feature(payment): Integrate paystack payment
create payments-paystack custom plugin

Commit:
feature(payment): Integrate paystack payment
create payments-paystack custom plugin
fix hound issues

Commit:
feature(payment): Integrate paystack payment
create payments-paystack custom plugin
fix hound issues
add hound.yml file

Commit:
feature(payment): Integrate paystack payment
create payments-paystack custom plugin
fix hound issues
add hound.yml file
modify eslintrc file

Commit:
feature(payment): Integrate paystack payment
create payments-paystack custom plugin
fix hound issues
add hound.yml file
modify eslintrc file

morekoya commented on .eslintrc:
```diff
@@ -27,7 +29,6 @@
         "destructuring": true,
         "forOf": true,
         "generators": false,
-        "modules": true,
```
Are you sure the removal of this isn't what is causing your import parsing error?

Felglitzzz replied:
No it isn't the reason

morekoya commented on .bashrc:
```diff
@@ -0,0 +1 @@
+ulimit -n 65536 65536
```
I'm pretty sure your `.bashrc` file should not have been pushed

Felglitzzz replied:
okay, I will fix it on my next push

morekoya commented on imports/plugins/custom/payments-paystack/client/checkout/paystack.js:
```diff
@@ -0,0 +1,122 @@
+          key: publicKey,
+          name: doc.payerName,
+          email: doc.payerEmail,
+          reference: Random.id(),
```
Isn't this supposed to be someway of tracking the product purchased? Won't it be better for the reference to be the product id?

Felglitzzz replied:
Yes, it is. Thank you for the feedback. Noted.

Felglitzzz replied:
The reference is the same as or say works as the product id

morekoya commented on imports/plugins/custom/payments-paystack/client/settings/components/paystackSettingsForm.js:
```diff
@@ -0,0 +1,134 @@
+
+          <div>
+            <Checkbox
+              label="Authorize"
```
I think that by the time a string is being repeated more three times, you might want to save it as a constant. This makes it easier to make make file wide changes when necessary.

Felglitzzz replied:
okay, noted

morekoya commented on imports/plugins/custom/payments-paystack/server/i18n/en.json:
```diff
@@ -0,0 +1,27 @@
+    "reaction-payments": {
+      "admin": {
+        "shortcut": {
+          "paystackPaymentProviderLabel": "paystack Payment",
```
Any particular reason why your phrases start with small letters?

Felglitzzz replied:
No reason, an oversight on my part

Commit:
feature(payment): Integrate paystack payment
create payments-paystack custom plugin
fix hound issues
add hound.yml file
modify eslintrc file

Commit:
feature(payment): Integrate paystack payment
create payments-paystack custom plugin
fix hound issues
add hound.yml file
modify eslintrc file
remove .bashrc file

Commit:
feature(payment): Integrate paystack payment
create payments-paystack custom plugin
fix hound issues
add hound.yml file
modify eslintrc file
remove .bashrc file
implement feedback from TTL

Commit:
feature(payment): Integrate paystack payment
create payments-paystack custom plugin
fix hound issues
add hound.yml file
modify eslintrc file
remove .bashrc file
implement feedback from TTL

Commit:
feature(payment): Integrate paystack payment
create payments-paystack custom plugin
fix hound issues
add hound.yml file
modify eslintrc file
remove .bashrc file
implement feedback from TTL

Commit:
feature(payment): Integrate paystack payment
create payments-paystack custom plugin
fix hound issues
add hound.yml file
modify eslintrc file
remove .bashrc file
implement feedback from TTL

Commit:
feature(payment): Integrate paystack payment
create payments-paystack custom plugin
fix hound issues
add hound.yml file
modify eslintrc file
remove .bashrc file
implement feedback from TTL

Commit:
feature(payment): Integrate paystack payment
create payments-paystack custom plugin
fix hound issues
add hound.yml file
modify eslintrc file
remove .bashrc file
implement feedback from TTL

Commit:
feature(payment): Integrate paystack payment
create payments-paystack custom plugin
fix hound issues
add hound.yml file
modify eslintrc file
remove .bashrc file
implement feedback from TTL

Commit:
feature(payment): Integrate paystack payment
create payments-paystack custom plugin
fix hound issues
add hound.yml file
modify eslintrc file
remove .bashrc file
implement feedback from TTL

Commit:
feature(payment): Integrate paystack payment
create payments-paystack custom plugin
fix hound issues
add hound.yml file
modify eslintrc file
remove .bashrc file
implement feedback from TTL

Commit:
feature(payment): Integrate paystack payment
create payments-paystack custom plugin
fix hound issues
add hound.yml file
modify eslintrc file
remove .bashrc file
implement feedback from TTL

Commit:
feature(payment): Integrate paystack payment
create payments-paystack custom plugin
fix hound issues
add hound.yml file
modify eslintrc file
remove .bashrc file
implement feedback from TTL

Commit:
feature(payment): Integrate paystack payment
create payments-paystack custom plugin
fix hound issues
add hound.yml file
modify eslintrc file
remove .bashrc file
implement feedback from TTL

Commit:
feature(payment): Integrate paystack payment
create payments-paystack custom plugin
fix hound issues
add hound.yml file
modify eslintrc file
remove .bashrc file
implement feedback from TTL

Commit:
feature(payment): Integrate paystack payment
create payments-paystack custom plugin
fix hound issues
add hound.yml file
modify eslintrc file
remove .bashrc file
implement feedback from TTL
add app.json for review app

Felglitzzz committed:
Merge branch 'develop' into feature/157244155/integrate-paystack-payment

Commit:
feature(payment): Integrate paystack payment
create payments-paystack custom plugin
fix hound issues
add hound.yml file
modify eslintrc file
remove .bashrc file
implement feedback from TTL
add app.json for review app

Commit:
Merge branch 'feature/157244155/integrate-paystack-payment' of https://github.com/andela/nazgul-rc into feature/157244155/integrate-paystack-payment

Commit:
feature(payment): Integrate paystack payment
create payments-paystack custom plugin
fix hound issues
add hound.yml file
modify eslintrc file
remove .bashrc file
implement feedback from TTL
add app.json for review app
fix approval and cancel button on order dashboard

Commit:
feature(payment): Integrate paystack payment
create payments-paystack custom plugin
fix hound issues
add hound.yml file
modify eslintrc file
remove .bashrc file
implement feedback from TTL
add app.json for review app
fix approval and cancel button on order dashboard
fix paystack unstability on heroku

Commit:
feature(payment): Integrate paystack payment
create payments-paystack custom plugin
fix hound issues
add hound.yml file
modify eslintrc file
remove .bashrc file
implement feedback from TTL
add app.json for review app
fix approval and cancel button on order dashboard
fix paystack unstability on heroku
